User component feautres user registration task - 2

RODID opened the pull request:
# RegisterNewUser - Pull Request

## Description

This pull request brings the features that allows a user to create a new user that registers into the program. fixed design and some minor bugs fixes.

## Changes Made

* RegisterNewUserPage [design and functions]
* RegisterUserDTO - [error messages for user friendly usage]

## Related Ticket or Issue

- Fixes #15 
- Closes #Blazor-App/15

## Checklist

- [x] Code adheres to the coding conventions
- [] Unit tests are added for new code
- [] Existing unit tests are updated if needed
- [x] Code follows the principles of Clean Architecture
- [] Code formatting complies with the `.editorconfig` file
- [] Pull request template is filled out

Commit:
update
- registrering funkar
-design
- error handling

Commit:
update
-Error handling

Commit:
update design and functions

Commit:
update for pull request

BojanMirkovic commented on BlazorApp/Layout/NavMenu.razor:
```diff
@@ -1,39 +1,48 @@
+    	<!--Registrating symbol changed using Bootstrap-->
+    	<div class="nav-item px-3">
+        	<NavLink class="nav-link" href="register">
+            	<i class="bi bi-person-plus fs-4" aria-hidden="true"></i> Create User
```
change HTML tag " i " to the "span" as it was done in other exampels , "span" can be used more universally for wrapping icons or other inline elements, while "i" may give the wrong impression that it is purely for text styling purposes.
I think that more suitable name is Sign Up instead for Create User.

RODID replied:
<i> has changed to <span>
The name of the create user to Sign up

BojanMirkovic commented on BlazorApp/wwwroot/index.html:
I dont understand why did you change index.html, it looks like you  are adding Bootstrap Icons and optional Bootstrap JavaScript, but it also introduces some unnecessary duplication of existing code. If the pupose of this is to add new icon for Sign Up in NavMenu it can be done on diferent way, all what you have to du is to add new icon in NavMenu.razor.css

RODID replied:
this is reverted to past code, one symbol is left for  sign up

BojanMirkovic committed:
I have added _fixture.Customize(new NoDataAnnotationsCustomization());
To ignore DataAnnotation from DTO and to fix workflow error

Commit:
update

BojanMirkovic committed:
removing AutoFixture that I have instaled

Commit:
Merge branch 'UserComponentFeautres-UserRegistrationTask' of https://github.com/InFiNet-Code-AB/Blazor-App into UserComponentFeautres-UserRegistrationTask-2

BojanMirkovic committed:
Fixing Test error message from workflow,
I remowed  test Handle_ValidNewUser_ReturnsRegisteredUser becouse test is doing the same thing as test  Handle_ValidNewUser_ReturnsRegisteredUser1

BojanMirkovic committed:
Merge branch 'UserComponentFeautres-UserRegistrationTask' of https://github.com/InFiNet-Code-AB/Blazor-App into UserComponentFeautres-UserRegistrationTask

BojanMirkovic commented on BlazorApp/wwwroot/css/bootstrap/RegisterPageStyleSheet.css:
You don't need this CSS. In this situation, you should use Bootstrap classes for styling instead.

RODID replied:
Done. this page is gone

Commit:
update!

BojanMirkovic commented on BlazorApp/wwwroot/index.html:
```diff
@@ -1,4 +1,3 @@
-<!DOCTYPE html>
```
The <!DOCTYPE html> declaration at the beginning of an HTML document is extremely important because it tells the browser how to interpret the document.

RODID replied:
added

BojanMirkovic commented on BlazorApp/Pages/RegisterNewUser.razor:
```diff
@@ -5,57 +5,118 @@
+                            </InputSelect>
+                        </div>
+
+                        <div class="form-group">
```
div class="form-group" is old class and is not in use anymore, Chat gpt is often making this mistacke. Instead use  mb-3  for setting a bottom margin. or mt-3 for setting a top margin.
in Bootstrap 5, the form-group class has been removed in favor of more utility classes and simplified form layouts. The same layout and spacing can now be achieved using margin utility classes, such as mb-3 (which means "margin-bottom: 3"), along with form-control for inputs.

RODID replied:
ändrade detta på alla inputfields

Commit:
update
